src/messaging/ReliableMessageContext.h

```diff
@@ -124,6 +124,10 @@ class ReliableMessageContext
     /// Determine whether this exchange is a EphemeralExchange for replying a StandaloneAck
     bool IsEphemeralExchange() const;
 
+    // For UpdateNOC command, I'm the last exchange, I must release the session when finishing my work.
+    void SetAutoReleaseSession();
+    bool IsAutoReleaseSession();
```

Comment:
```suggestion
    // If SetAutoReleaseSession() is called, this exchange must be using a SecureSession, and should
    // evict it when the exchange is done with all its work (including any MRP traffic).
    void SetAutoReleaseSession();
    bool ReleaseSessionOnDestruction();
```
(Yes, that makes the setter and getter asymmetric, but the getter name accurately reflects what we are actually asking this thing to do.  I don't have a strong opinion about the setter name.)

I could also live with `AutoReleaseSession` or `GetAutoReleaseSession`, I guess, but "IsAutoReleaseSession" just doesn't sound right.